app/src/main/java/com/lagradost/cloudstream3/syncproviders/providers/GithubApi.kt

```diff
@@ -0,0 +1,140 @@
+package com.lagradost.cloudstream3.syncproviders.providers
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.lagradost.cloudstream3.*
+import com.lagradost.cloudstream3.AcraApplication.Companion.getKey
+import com.lagradost.cloudstream3.AcraApplication.Companion.setKey
+import com.lagradost.cloudstream3.mvvm.logError
+import com.lagradost.cloudstream3.syncproviders.AuthAPI
+import com.lagradost.cloudstream3.syncproviders.InAppAuthAPI
+import com.lagradost.cloudstream3.syncproviders.InAppAuthAPIManager
+import com.lagradost.cloudstream3.utils.AppUtils.parseJson
+import com.lagradost.cloudstream3.utils.AppUtils.toJson
+import com.lagradost.cloudstream3.utils.AppUtils.tryParseJson
+import com.lagradost.cloudstream3.utils.Coroutines.ioSafe
+import com.lagradost.nicehttp.RequestBodyTypes
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.transport.URIish
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.attribute.FileAttribute
+
+
+class GithubApi(index: Int) : InAppAuthAPIManager(index){
+    override val idPrefix = "Github"
+    override val name = "Github"
+    override val icon = R.drawable.ic_github_logo
+    override val requiresPassword = true
+    override val createAccountUrl = "https://github.com/settings/tokens/new"
+
+    data class GithubOAuthEntity(
+        var repoUrl: String,
+        var token: String
+    )
+    companion object {
+        const val GITHUB_USER_KEY: String = "github_user" // user data like profile
+        var currentSession: GithubOAuthEntity? = null
+    }
+    private fun getAuthKey(): GithubOAuthEntity? {
+        return getKey(accountId, GITHUB_USER_KEY)
+    }
+    private class repodata (
+        @JsonProperty("full_name") val repoUrl: String
+    )
+
+
+    private suspend fun initLogin(githubToken: String): Boolean{
+        val response = app.post("https://api.github.com/user/repos",
+            headers= mapOf(
+                Pair("Accept" , "application/vnd.github+json"),
+                Pair("Authorization", "token $githubToken"),
```

Comment:
FYI you can do "Authorization" to "token $githubToken", it is just easier to write impo than pair